### DISCUSSION TITLE
2D random walk Problem

### DISCUSSION THREAD

**Discussion by vermaprk** [idaholab/moose#23886] [Q&A Modules: General] [answered]
Hi
I am looking to model the Brownian motion of say N particles in a 2d domain. How can I implement such model using moose framework. I have simulated such problem in MATLAB and python but how can I implement it in moose? Also I want to implement some rules for further guided movement of particles. 
If there are some existing resources please tag here.

Thanks

**Comment by GiudGiud** [comment, marked as answer]
Hello

I think you will want to use the ray tracing module to do this
We had started a project on something similar but never finished, please feel free to examine the latest commits here:
https://github.com/loganharbour/moose/tree/rt_mpm

Guillaume

**Reply by vermaprk** [reply]
can I run a python script (for the random walk) within the moose input file. So that at each time step the python script provides me the position of particles which can be further used by the moose variables and material properties.

**Reply by GiudGiud** [reply]
No you can't run python within MOOSE right now. 

If the logic in the python script is not too complicated, it should be transcribed in the RayKernels / RayTracingStudy objects in the module. This will ensure the most performance as well

**Reply by hugary1995** [reply]
> No you can't run python within MOOSE right now.

well, the user can always write a userobject to do anything... in this case running a python script.

**Reply by GiudGiud** [reply]
Is it time for the `RunCommand` user object?
It's definitely not preferred at this point

**Reply by hugary1995** [reply]
I don't prefer that either. I was just pointing out the possibility.

**Reply by vermaprk** [reply]
> If the logic in the python script is not too complicated, it should be transcribed in the RayKernels / RayTracingStudy objects in the module. This will ensure the most performance as well

OK. Thanks
Will be posting further upon transcribing the python code into RayKernels.
